applications/nrf_desktop/doc/porting_guide.rst

```diff
@@ -1,10 +1,157 @@
 .. _porting_guide:
 
-Application porting guide
-#########################
+nRF Desktop application porting guide
+#####################################
+
+This porting guide for the nRF Desktop project describes how to adapt the nRF Desktop application to different hardware.
+It describes the configuration sources that are used for the default configuration, and lists steps required for adding a new board.
+
+Configuration sources
+*********************
+
+The nRF Desktop project uses the following files as sources of its configuration:
+
+    * DTS files -- reflect the hardware configuration (see :ref:`device-tree`).
+    * ``_def`` files -- contain configuration arrays for the application modules and are specific to the nRF Desktop project.
+    * Kconfig files -- reflect the software configuration (see :ref:`kconfig_tips_and_tricks`).
+
+You must modify these configuration sources when `Adding a new board`_.
+For information about differences between DTS files and Kconfig, see `Devicetree vs Kconfig`_.
+
+Board configuration
+*******************
+
+The nRF Desktop application is modular.
+Depending on requested functions, it can provide mouse, keyboard, or dongle functionality.
+The selection of modules depends on the chosen role and also on the selected board.
+For more information about modules available for each configuration, see :ref:`nrf_desktop_architecture`.
+
+For a board to be supported by the application, you must provide a set of configuration files at ``applications/nrf_desktop/configuration/your_board_name``.
+The application configuration files define both a set of options with which the nRF Desktop project will be created for your board and the selected :ref:`nrf_desktop_build_types`.
+Include the following files in this directory:
+
+Mandatory configuration files
+    * Application configuration file for the ``ZDebug`` build type.
+    * Configuration files for the selected modules.
+
+Optional configuration files
+    * Application configuration files for other build types.
+    * Configuration file for the bootloader.
+    * DTS overlay file.
```

Comment:
Partition manager configuration file for MCUBoot configurations (added recently)